options.go

```diff
@@ -15,7 +15,7 @@ import (
 // UpdateCallback is invoked upon arrival of a message for a table partition.
 // The partition storage shall be updated in the callback.
 // Headers can converted to a map using FromSarama function.
-type UpdateCallback func(s storage.Storage, partition int32, key string, value []byte, headers ...*sarama.RecordHeader) error
+type UpdateCallback func(s storage.Storage, partition int32, key string, value []byte, headers Headers) error
```

Comment:
Thanks, awesome! Only thing is, that we're now converting the headers on all message, even though we might never use them. If we'd have a new context that wraps the `sarama.ConsumerMessage` passed to the update callback
```go

type UpdateContext interface{
   Headers() Headers
}

// then pass this instead of the headers
type UpdateCallback func(s storage.Storage, partition int32, key string, value []byte, updCtx UpdateContext) error
```
then we could transform the headers only if we need them. Also we would be flexible in case we want to add more fields at some point, like Offset or Timestamp and not break the whole API by just adding a new function in the interface.

But of course it's more work and I don't want to load that on you and I'm happy to support you there :). 
Still though - do you think it's worth it or is the use case so hypothetical that we can live with the performance-overhead (if any) and the extra API-change and leave it as is?

Reply:
Good catch!

I like the idea of introducing `UpdateContext` considering this is already a breaking change. I think it makes sense to wrap other arguments in the context as well:
```go
type UpdateContext interface {
  Storage() storage.Storage
  Key() string
  Value() []byte
  Partition() int32
  Headers() Headers
}

type UpdateCallback func(ctx UpdateContext)
```

Thoughts?

Reply:
Awesome, love it. Could even also "Topic()", since the rest of the fields are redundant too.
Many thanks for your work!

Reply:
Thanks!
Done.